### PR TITLE
MARP-228 Problem with meta: proc connector

### LINKED ISSUE #9: Login Token Refresh
Hi @ivy-sgi ,

we are currently trying to setup the Metaproc-Connector for our RPA demos.

In addition to some problems on the Metaproc side, we also noticed that there seems to be a refresh problem with the token of the RPA interface to Metaproc on the Axon Ivy side. The auth token feature implementation seems to somehow cache the login but when the token expires it is not refreshed. 
(Feedback Metaproc developer: "the token in the HTTP logs in the AxonIvy designer is the same for every execution. However, when I call up the login using postman, for example, I always get a different new token. When I restart the AxonIvy app/designer, the first execution always seems to work.")

Can you please revise the Auth implementation here? Possibly after consultation with Ivyteam...

Thx & br
Thomas

### PR DESCRIPTION
**Pull request by tutn-axonivy**

### CODE REVIEW

File: metaproc-connector/src/com/axonivy/connector/metaproc/auth/AuthTokenFeature.java [hunk 1]
```diff
@@ -21,6 +21,7 @@ public class AuthTokenFeature implements Feature {
   private static final String LOGIN_PATH = "/metarest/user/login";
 
   public static String TOKEN;
+  public static long TOKEN_EXPIRED_DURATION = 0;
```
Comment (ghost) [inline metaproc-connector/src/com/axonivy/connector/metaproc/auth/AuthTokenFeature.java]: rename TOKEN_EXPIRED_DURATION to TOKEN_EXPIRED_TIME

**Review by ghost — APPROVED** [review]

File: metaproc-connector-demo/config/variables.yaml [hunk 2]
```diff
@@ -3,3 +3,4 @@ Variables:
     Url: http://localhost:8080
     Username: admin
     Password: 1234
+    TokenExpiredDuration: 3600000 # Milisecond
```
Comment (ivy-rew) [inline metaproc-connector-demo/config/variables.yaml:6]: this is an invalid and broken yaml: comments should reside on top of the variable name
https://dev.axonivy.com/doc/10.0/designer-guide/configuration/variables.html#variables

Comment (ivy-rew) [reply metaproc-connector-demo/config/variables.yaml:6]: Also I think using milliseconds here is not what the customer wants,
A duration of seconds or better minutes is much easier to setup

File: metaproc-connector/src/com/axonivy/connector/metaproc/auth/AuthTokenFeature.java [hunk 3]
```diff
@@ -52,7 +53,10 @@ public void filter(ClientRequestContext context) throws IOException {
               .accept(MediaType.APPLICATION_JSON).post(entity)) {
         var node = response.readEntity(ObjectNode.class);
         TOKEN = node.get("loginResult").get("token").asText();
+        TOKEN_EXPIRED_TIME = System.currentTimeMillis() + Long.valueOf(Ivy.var().get("metaproc.TokenExpiredDuration"));
         context.getHeaders().add(TOKEN_HEADER, TOKEN);
+      } catch (Exception e) {
+        Ivy.log().error(e.getMessage());
```
Comment (ivy-rew) [inline metaproc-connector/src/com/axonivy/connector/metaproc/auth/AuthTokenFeature.java:59]: with this log, the stacktrace is swallowed ... with no trace what lead us here. 
also it's common sense in productive java code to enrich the message being logged with the local context

File: metaproc-connector-demo/config/variables.yaml (hunk 2 above)
Comment (tutn-axonivy) [reply metaproc-connector-demo/config/variables.yaml:6]: Hi @ivy-rew, so the variable should start with lower case right?

Comment (ivy-rew) [reply metaproc-connector-demo/config/variables.yaml:6]: No, as you see there are other variables starting with capital letters. So I'd keep this pattern.

**Pull request merged**